src/core/runtime.d

```diff
@@ -37,9 +37,58 @@ extern(C) int rt_init();
 /// C interface for Runtime.terminate, returns 1/0 instead of bool
 extern(C) int rt_term();
 
+/**
+ * This type is returned by the module unit test handler to indicate testing
+ * results.
+ */
+struct UnitTestResult
+{
+    /**
+     * Number of modules which were tested
+     */
+    size_t executed;
+
+    /**
+     * Number of modules passed the unittests
+     */
+    size_t passed;
+
+    /**
+     * Should the main function be run or not. Should be ignored if any tests
+     * failed.
+     */
+    bool runMain;
+
+    /**
+     * Should we print a summary of the results. Ignored if 0 tests executed.
+     */
+    bool summarize;
```

Comment:
I'm not sure what the summary actually contains but many frameworks print the number of tests run even if 0 tests were run.

Reply:
OK, so D's determination if we are in "unittest mode" is whether there exists any unit tests at all. So if there are no unit tests run, the runtime thinks that you are not testing.

But thinking about this some more, we can easily add an option for "don't ever run main, even without unit tests" as well, as we are already processing a runtime switch. I'll add it, and let you know.

BTW, the summary is done in the `dmain2` file, I added it in this PR, look for the `printf` statements